### DISCUSSION TITLE
[mixed time-series] extra x-axis labels

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#18495] [Ideas]
When creating the chart with only 1-2 bars on x-axis, some unnecessary x-axis labels are displayed.

### Expected results

Number of x-axis labels should be same as number of data points.

### Actual results

Excessive x-axis labels are displayed.

#### Screenshots
With Time Grain: Month
![image](https://user-images.githubusercontent.com/74634977/125559617-5a9ee615-081b-48e3-9a0b-8ba4c29e5154.png)

With Time Grain: Quarter
![image](https://user-images.githubusercontent.com/74634977/125559703-2ca06b50-de0b-4b0a-9c19-77798c17dde3.png)

#### How to reproduce the bug

1. Create a Mixed Time-series chart with limited number of points on x-axis

### Environment

- superset version: `1.2`
- python version: `3.7`
- node.js version: `v14.15.5`

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by junlincc** [comment]
> some unnecessary x-axis labels are displayed.

you mean these? what time grain did you set for the bar? 
![125559617-5a9ee615-081b-48e3-9a0b-8ba4c29e5154](https://user-images.githubusercontent.com/67837651/125705285-9dc62a0f-64c2-4f02-a752-ee4de798ed0d.png)

**Comment by kamalkeshavani-aiinside** [comment]
Yes, this was with Time grain 'month'. I updated the original comment with this info.

**Comment by junlincc** [comment]
Got it, this is a limitation from Echarts, the charting library we use. unfortunately there's no easy way to resolve ATM. apologies🙏

**Comment by Pedrowach** [comment]
This bug is still Active. Any update to solve it? Its really annoying cause the data points are no correctly correlated to the X-axis labels.

**Reply by JoseAnaya28** [reply]
Agreed, I want to display a weekly chart but the charts display labels for each day confusing the end users. Super annoying 

<img width="534" alt="image" src="https://github.com/apache/superset/assets/60910800/36db78d7-039d-427d-b574-bbf68aacfd9a">

Should only appear two labels for each week's data point

**Comment by Caillou33** [comment]
Hello. No news on this matter ??

**Reply by hoang1233** [reply]
how do you set 2020 - Q3 for x_label, i'm finding solution for this
